### PR TITLE
Unnest Scene and Segue structs to work around a LLVM bug

### PR DESCRIPTION
**Pull request by ryanbooker**
Works around https://github.com/AliSoftware/SwiftGen/issues/57

I've left the default Segue and Scene names as is. We can't use StoryboardScene and StoryboardSegue as they conflict with the protocols. I'm not attached to a particular name if anyone has something better.

### CODE REVIEW

**Comment by AliSoftware** [comment]
Thx for the PR!

Could you please add an entry in the CHANGELOG about that change too?

**Comment by AliSoftware** [comment]
It might be worth renaming the protocols `StoryboarfSceneType` and `StoryboardSegueType` so that we can then use`StoryboardScene` and `StoryboardSegue` as the default names.

**Comment by ryanbooker** [comment]
Should I update the version to 0.7.3?

**Comment by AliSoftware** [comment]
Yeah was wondering that as well. In cases like SwiftGen it's hard to do semantic versioning properly because what changed is actually only a template. People using custom templates already won't be affected by that change and the SwiftGen API itself (the API of the command line and the API of the framework) didn't change, only the template did.
I'm never sure if the server in that case should apply only to the SwiftGen & GenumKit.framework APIs or to the generated code as well.

I'd say it's ok to only bump to 0.7.3 anyway as SwiftGen is still in 0.x.y and major version 0 can always break anyway.

File: README.md [hunk 1]
```diff
@@ -238,17 +238,17 @@ extension StoryboardScene where Self: RawRepresentable, Self.RawValue == String
...
+protocol StoryboardSegueType : RawRepresentable { }
 
 extension UIViewController {
-  func performSegue<S : StoryboardSegue where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
+  func performSegue<S : StoryboardSegueType where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
     performSegueWithIdentifier(segue.rawValue, sender: sender)
   }
 }
 
 extension UIStoryboard {
```
Comment (AliSoftware) [inline README.md]: This should be updated as well

File: SwiftGen.playground/Pages/Storyboards-Demo.xcplaygroundpage/Contents.swift [hunk 2]
```diff
@@ -39,17 +39,17 @@ extension StoryboardScene where Self: RawRepresentable, Self.RawValue == String
...
+protocol StoryboardSegueType : RawRepresentable { }
 
 extension UIViewController {
-  func performSegue<S : StoryboardSegue where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
+  func performSegue<S : StoryboardSegueType where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
     performSegueWithIdentifier(segue.rawValue, sender: sender)
   }
 }
 
 extension UIStoryboard {
```
Comment (AliSoftware) [inline SwiftGen.playground/Pages/Storyboards-Demo.xcplaygroundpage/Contents.swift]: Ditto

File: SwiftGen.playground/Pages/Storyboards-Demo.xcplaygroundpage/Contents.swift [hunk 3]
```diff
@@ -75,16 +75,16 @@ extension UIStoryboard {
...
-            case ShowPassword = "ShowPassword"
+            case ShowPassword = "ShowP//: #### Usage Example
+assword"
         }
     }
 }
 
-//: #### Usage Example
 
 let initialVC = UIStoryboard.Scene.Wizard.initialViewController()
```
Comment (AliSoftware) [inline SwiftGen.playground/Pages/Storyboards-Demo.xcplaygroundpage/Contents.swift]: Update

File: UnitTests/expected/Storyboards-Message-Lowercase.swift.out [hunk 4]
```diff
@@ -26,47 +26,45 @@ extension StoryboardScene where Self: RawRepresentable, Self.RawValue == String
...
+  func performSegue<S : StoryboardSegueType where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
     performSegueWithIdentifier(segue.rawValue, sender: sender)
   }
 }
 
-extension UIStoryboard {
-  struct Scene {
-    enum Message : String, StoryboardScene {
-      static let storyboardName = "Message"
+struct Scene {
```
Comment (AliSoftware) [inline UnitTests/expected/Storyboards-Message-Lowercase.swift.out]: Replace with StoryboardScene (here and in other places; same for Segue of course)

File: SwiftGen.playground/Pages/Storyboards-Demo.xcplaygroundpage/Contents.swift [hunk 5]
```diff
@@ -117,6 +117,4 @@ segues for a specific storyboard.
 //  case .Custom:
 //    // Prepare for your custom segue transition
 //  case .NonCustom:
-//    // Pass in information to the destination View Controller
-//  }
-//}
+//    // Pass in information to the destinatio
```
Comment (AliSoftware) [inline SwiftGen.playground/Pages/Storyboards-Demo.xcplaygroundpage/Contents.swift]: ? Why has this been truncated?

Comment (ryanbooker) [inline SwiftGen.playground/Pages/Storyboards-Demo.xcplaygroundpage/Contents.swift]: No idea. Let me look. I didn't touch the playground until just now with these comments.

**Comment by ryanbooker** [comment]
I've made all your changes. Cheers.

File: README.md [hunk 6]
```diff
@@ -238,52 +238,50 @@ extension StoryboardScene where Self: RawRepresentable, Self.RawValue == String
...
-      case Custom = "Custom"
-      case NonCustom = "NonCustom"
+    case ValidatePassword = "Validate_Password"
+    static func validatePasswordViewController() -> UIViewController {
+        return Wizard.ValidatePassword.viewController()
     }
   }
 }
+
+struct Segue {
```
Comment (AliSoftware) [inline README.md]: `StoryboardSegue`

File: README.md [hunk 7]
```diff
@@ -238,52 +238,50 @@ extension StoryboardScene where Self: RawRepresentable, Self.RawValue == String
...
+  func performSegue<S : StoryboardSegueType where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
     performSegueWithIdentifier(segue.rawValue, sender: sender)
   }
 }
 
-extension UIStoryboard {
-  struct Scene {
-    enum Message : String, StoryboardScene {
-      static let storyboardName = "Message"
+struct Scene {
```
Comment (AliSoftware) [inline README.md]: `StoryboardScene`

**Comment by AliSoftware** [comment]
Looks good aside from the two you missed in the README :wink:

**Comment by iGranDav** [comment]
Just tested this PR in my project on the same issue, looks perfect to me thanks @ryanbooker :+1:

**Comment by AliSoftware** [comment]
I don't have my homebrew fork and scripts in my work computer, will make a homebrew release tonite.

**Comment by ryanbooker** [comment]
Great. Glad to have helped.

**Comment by AliSoftware** [comment]
SwiftGen 0.7.3 is out, just got merged in Homebrew :smiley:
Thanks @ryanbooker for this :+1:

**Pull request merged**